### DISCUSSION TITLE
renovate-config-validator can't parse `.renovaterc`

### DISCUSSION THREAD

**Discussion by suzuki-shunsuke** [renovatebot/renovate#22990] [Request Help] [answered]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

35.143.0

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

Placeholder value, please select the correct response from the dropdown

### Describe the problem

This is a bug report of `renovate-config-validator`.

I tested `renovate-config-validator` in my laptop, then I found `renovate-config-validator` didn't work if the file name is `.renovaterc`.

I installed `renovate-config-validator` by npm.

```console
$ npm i -g renovate
```

```console
$ renovate --version
35.143.0
```

This issue occurs with the version `35.41.0` too.

If the file name is `renovate.json`, `renovate-config-validator` works fine.

renovate.json

```json
{
  "extends": [
    "config:base",
    "helpers:pinGitHubActionDigests"
  ],
  "packageRules": [
    {
      "matchManagers": [
        "gomod"
      ],
      "postUpdateOptions": []
    }
  ]
}
```

```console
$ renovate-config-validator   
 INFO: Validating renovate.json
 INFO: Config validated successfully
```

But if I renamed `renovate.json` to `.renovaterc`, renovate-config-validator failed.

```console
$ mv renovate.json .renovaterc
```

```console
$ renovate-config-validator
 WARN: .renovaterc could not be parsed
       "err": {
         "message": "Unsupported file type",
         "stack": "Error: Unsupported file type\n    at getParsedContent (/Users/shunsukesuzuki/.volta/tools/image/packages/renovate/lib/node_modules/renovate/lib/workers/global/config/parse/file.ts:31:13)\n    at /Users/shunsukesuzuki/.volta/tools/image/packages/renovate/lib/node_modules/renovate/lib/config-validator.ts:83:53"
       }
```

## minimal reproduction repository

https://github.com/suzuki-shunsuke/renovate-reproduction-repo-1

https://github.com/suzuki-shunsuke/renovate-reproduction-repo-1/actions/runs/5383306525/jobs/9769886633

### Relevant debug logs

```console
$ LOG_LEVEL=debug renovate-config-validator 
```

<details><summary>Logs</summary>

```
DEBUG: Using RE2 as regex engine
 WARN: File could not be parsed
       "file": ".renovaterc",
       "err": {
         "message": "Unsupported file type",
         "stack": "Error: Unsupported file type\n    at getParsedContent (/Users/shunsukesuzuki/.volta/tools/image/packages/renovate/lib/node_modules/renovate/lib/workers/global/config/parse/file.ts:31:13)\n    at /Users/shunsukesuzuki/.volta/tools/image/packages/renovate/lib/node_modules/renovate/lib/config-validator.ts:87:53"
       }
DEBUG: Checking for config file in /Users/shunsukesuzuki/repos/src/github.com/suzuki-shunsuke/test-github-action/config.js
DEBUG: No config file found on disk - skipping
```

</details>

### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction in the description above

**Comment by suzuki-shunsuke** [comment]
This discussion came from bug reports of my OSS.

- https://github.com/suzuki-shunsuke/github-action-renovate-config-validator/issues/398
- https://github.com/suzuki-shunsuke/github-action-renovate-config-validator/issues/399

**Reply by suzuki-shunsuke** [reply]
We can avoid the issue by renaming the file to the other name such as `.renovaterc.json5`.

**Reply by suzuki-shunsuke** [reply]
What's the expected file format of `.renovaterc`?
I thought it was JSON or JSON5, but this isn't documented so maybe this is my misunderstanding.

**Reply by suzuki-shunsuke** [reply]
https://github.com/renovatebot/renovate/blob/ef6f0c7ffa2636c19e61932c4e8195fa01fde2c3/lib/workers/repository/init/merge.ts#L132-L195

It seems Renovate expects `JSON`.

https://www.npmjs.com/package/upath

**Comment by suzuki-shunsuke** [comment]
It seems `.renovaterc` isn't allowed.

https://github.com/renovatebot/renovate/blob/ef6f0c7ffa2636c19e61932c4e8195fa01fde2c3/lib/workers/global/config/parse/file.ts#L11-L33

**Comment by suzuki-shunsuke** [comment]
Renovate can parse `.renovaterc`.

https://github.com/suzuki-shunsuke/renovate-reproduction-repo-1/pull/2

```
DEBUG: Found .renovaterc config file
DEBUG: Repository config
{
  "fileName": ".renovaterc"
  "config": {
    "extends": [
      "config:base",
      "helpers:pinGitHubActionDigests"
    ],
    "packageRules": [
      {
        "matchManagers": [
          "gomod"
        ],
        "postUpdateOptions": []
      }
    ]
  }
}
```

But `renovate-config-validator` doesn't work.

**Comment by suzuki-shunsuke** [comment]
I created a pull request to resolve the issue.

- https://github.com/renovatebot/renovate/pull/22991

**Reply by suzuki-shunsuke** [reply, marked as answer]
This issue was solved by https://github.com/renovatebot/renovate/pull/22991 .

**Comment by suzuki-shunsuke** [comment]
I guess the validation code of `renovate-config-validator` is different from the code of `renovate`.
They should be unified.

**Reply by viceice** [reply]
yes, we should unify them to use as much as possible same code

**Reply by rarkins** [reply]
I was thinking `renovate --dry-run=config`
